handshake_random.go

```diff
@@ -15,13 +15,13 @@ type handshakeRandom struct {
 	randomBytes [randomBytesLength]byte
 }
 
-func (h *handshakeRandom) Marshal() ([]byte, error) {
-	out := make([]byte, handshakeRandomLength)
+func (h *handshakeRandom) marshalFixed() [handshakeRandomLength]byte {
+	var out [handshakeRandomLength]byte
 
 	binary.BigEndian.PutUint32(out[0:], uint32(h.gmtUnixTime.Unix()))
 	copy(out[4:], h.randomBytes[:])
 
-	return out, nil
+	return out
 }
 
 func (h *handshakeRandom) Unmarshal(data []byte) error {
```

Comment:
The `Marshal` is replaced with `marshalFixed` but `Unmarshal` is still here. Does some code still use `Unmarshal`? And if so, why doesn't it need to use the new `unmarshalFixed` func?

Reply:
https://github.com/pion/dtls/blob/3caecff8298b6b091c45a1914e9bf54f3b2a7e7e/handshake_message_client_hello.go#L68-L70
https://github.com/pion/dtls/blob/3caecff8298b6b091c45a1914e9bf54f3b2a7e7e/handshake_message_server_hello.go#L69-L71
these two functions unmarshal random into Slice, and Go Slice can't be converted to Array without copy.

Reply:
These functions are not called frequently, so copy may be fine.

Reply:
Removed `handshakeRandom.Unmarshal`